services/authorization-service.js

```diff
@@ -0,0 +1,86 @@
+const ERROR = require('../constants/error-constants');
+
+class AuthorizationService {
+
+    constructor(configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    /**
+     * Function to get a user's scopes for a given permission
+     * @param user - the current user
+     * @param permission - the name of the permission without scopes, please use a permissions constant for this input
+     * Example: "entity:action"
+     * @returns {*|string[]|*[]} - an array containing the scopes for the permission or an empty array, all scopes will be lowercase
+     */
+    getPermissionScope(user, permission){
+        const userPermissions = user?.permissions;
+        if (!userPermissions || !permission){
+            return [];
+        }
+        // Loop through the user's permissions until one matching the input permission is found
+        for (const userPermission of userPermissions){
+            let permissionAndScope = parsePermissionString(userPermission);
```

Comment:
I think permissionAndScope should be constant?

Reply:
This has been updated to be a constant